lib/ensemble_app.dart

```diff
@@ -72,12 +72,38 @@ class EnsembleAppState extends State<EnsembleApp> {
 
         if (snapshot.hasError) {
           return _appPlaceholderWrapper(
-              widget: ErrorScreen(LanguageError(
-                "Error loading configuration",
-                detailError: snapshot.error.toString()
-              )
-            )
-          );
+              loadingBackgroundColor:
+              const Color(0XFF000000).withOpacity(0.9),
+              widget: AlertDialog(
+                backgroundColor: const Color(0xff011A23),
+                actions: [
+                  TextButton(
+                      onPressed: () {
+                        Navigator.of(context).popUntil((route) => route.isFirst);
+                      },
+                      child: const Text(
+                        "OK",
+                        style: TextStyle(
+                          color: Color(0xff059ACD),
+                        ),
+                      ))
+                ],
+                title: const Text(
+                  "Invalid QR code",
```

Comment:
ensemble is a library, it doesn't know anything about QR Codes. It gets an appid and tries to load it. You need to check the error and see if it is due to definition not found and then show a message. Also every text within Ensemble must be translatable and must not be hardcoded.

Reply:
Mostly when we try to scan invalid qr code then config will not be loaded and it gives us error loading configuration So i have made an invalid qr code error message there but when we write invalid app id it gives us invalid page defination error so i appear invalid app id error there but if you want i will change this invalid qr code to invalid app id error message
@kmahmood74